code/iaas/auth-logic/src/main/java/io/cattle/platform/iaas/api/auth/dynamic/DynamicSchemaFactory.java

```diff
@@ -61,12 +63,14 @@ public Schema getSchema(Class<?> clz) {
 
     @Override
     public Schema getSchema(String type) {
+        if (type == null) {
```

Comment:
I agree about the cache. If we need it later. Up to you @cjellick  and @ibuildthecloud